syft/pkg/cataloger/common/cpe/java.go

```diff
@@ -184,6 +184,113 @@ func GroupIDsFromJavaPackage(p pkg.Package) (groupIDs []string) {
 	return GroupIDsFromJavaMetadata(p.Name, metadata)
 }
 
+// GroupIDFromJavaPackage returns the authoritative group ID for a Java package.
+// The order of precedence is:
+// 1. The group ID from the POM properties
+// 2. The group ID from the POM project
+// 3. The group ID from a select map of known group IDs
+// 3. The group ID from the Java manifest
+func GroupIDFromJavaMetadata(pkgName string, metadata pkg.JavaMetadata) (groupID string) {
```

Comment:
all of these new functions probably don't belong in the `cpe` package since they have been replicated to not affect the CPE generation logic today. My vote would be to put this under the java cataloger package an unexport all of the functions (with one exception: I'd duplicate the `looksLikeGroupID` function instead of trying to share it... since there isn't a specific definition for this and the impl may change based on the use case).

Reply:
I think I parsed this comment correctly - check [f1e05da](https://github.com/anchore/syft/pull/2033/commits/f1e05dace2ab23c476d5ecc6b65d4b363e4097d2)

If there are some edits or things to move around from here we can do a quick sync on the package organization and exports/imports